fdbrpc/FlowTests.actor.cpp

```diff
@@ -1271,3 +1272,115 @@ TEST_CASE("/flow/DeterministicRandom/SignedOverflow") {
+	// get
+	{
+		auto resultFuture = pks.get(NULL, LiteralStringRef("\xff\xff/cat/small0000000009"));
+		ASSERT(resultFuture.isReady());
+		auto result = resultFuture.getValue().get();
+		ASSERT(result == pkr1.getKeyValueForIndex(9).value);
+		auto emptyFuture = pks.get(NULL, LiteralStringRef("\xff\xff/cat/small0000000010"));
+		ASSERT(emptyFuture.isReady());
+		auto emptyResult = emptyFuture.getValue();
+		ASSERT(!emptyResult.present());
+	}
+	// general getRange
+	{
+		KeySelector start = KeySelectorRef(LiteralStringRef("\xff\xff/elepant"), false, -9);
+		KeySelector end = KeySelectorRef(LiteralStringRef("\xff\xff/frog"), false, +11);
+		auto resultFuture = pks.getRange(NULL, start, end, GetRangeLimits());
+		ASSERT(resultFuture.isReady());
+		auto result = resultFuture.getValue();
+		ASSERT(result.size() == 20);
+		ASSERT(result[0].key == pkr2.getKeyForIndex(90)) ;
+		ASSERT(result[result.size()-1].key == pkr3.getKeyForIndex(9));
+	}
+	// KeySelector points outside
+	{
+		KeySelector start = KeySelectorRef(pkr3.getKeyForIndex(999), true, -1110);
+		KeySelector end = KeySelectorRef(pkr1.getKeyForIndex(0), false, +1112);
+		auto resultFuture = pks.getRange(NULL, start, end, GetRangeLimits());
+		ASSERT(resultFuture.isReady());
+		auto result = resultFuture.getValue();
+		ASSERT(result.size() == 1110);
+		ASSERT(result[0].key == pkr1.getKeyForIndex(0)) ;
+		ASSERT(result[result.size()-1].key == pkr3.getKeyForIndex(999));
+	}
+	// GetRangeLimits with row limit
+	{
+		KeySelector start = KeySelectorRef(pkr2.getKeyForIndex(0), true, 0);
+		KeySelector end = KeySelectorRef(pkr3.getKeyForIndex(0), false, 0);
+		auto resultFuture = pks.getRange(NULL, start, end, GetRangeLimits(2));
+		ASSERT(resultFuture.isReady());
+		auto result = resultFuture.getValue();
+		ASSERT(result.size() == 2);
+		ASSERT(result[0].key == pkr2.getKeyForIndex(0));
+		ASSERT(result[1].key == pkr2.getKeyForIndex(1));
+	}
+	// GetRangeLimits with byte limit
+	{
+		KeySelector start = KeySelectorRef(pkr2.getKeyForIndex(0), true, 0);
+		KeySelector end = KeySelectorRef(pkr3.getKeyForIndex(0), false, 0);
+		auto resultFuture = pks.getRange(NULL, start, end, GetRangeLimits(10, 100));
+		ASSERT(resultFuture.isReady());
+		auto result = resultFuture.getValue();
+		int bytes = 0;
+		for (int i = 0; i < result.size()-1; ++i)
+			bytes +=  8 + pkr2.getKeyValueForIndex(i).expectedSize();
+		ASSERT(bytes < 100);
+		ASSERT(bytes + 8 + pkr2.getKeyValueForIndex(result.size()).expectedSize() >= 100);
+	}
+	// reverse test
+	{
+		KeySelector start = KeySelectorRef(pkr2.getKeyForIndex(0), true, 0);
+		KeySelector end = KeySelectorRef(pkr3.getKeyForIndex(0), false, 0);
+		auto resultFuture = pks.getRange(NULL, start, end, GetRangeLimits(100), false, true);
+		ASSERT(resultFuture.isReady());
+		auto result = resultFuture.getValue();
+		for (int i = 0; i < result.size(); ++i)
+			ASSERT(result[i] == pkr2.getKeyValueForIndex(result.size() - 1 - i));
+	}
```

Comment:
Add a test for registering overlapping ranges

Reply:
Done. Thanks.